File: backends/java/zappy-application/src/main/java/com/zappymart/application/cart/ViewCart.java
package com.zappymart.application.cart;

import com.zappymart.application.Visitor;
import com.zappymart.domain.cart.Cart;

public final class ViewCart {

    private final CurrentCart currentCart;

    public ViewCart(CurrentCart currentCart) {
        this.currentCart = currentCart;
    }

    public Cart execute(Visitor visitor) {
        return currentCart.forVisitor(visitor);
    }
}

File: backends/kotlin/zappy-application/src/main/kotlin/nl/zappymart/application/cart/ViewCart.kt
package nl.zappymart.application.cart

import nl.zappymart.application.Visitor
import nl.zappymart.domain.cart.Cart

class ViewCart(
    private val visitorCart: VisitorCart,
    private val cartPromotion: CartPromotion,
) {

    fun execute(visitor: Visitor): Cart = cartPromotion.refreshed(visitorCart.of(visitor))
}
